cli/command/image/build.go

```diff
@@ -247,6 +253,66 @@ func runBuild(dockerCli *command.DockerCli, options buildOptions) error {
 		buildCtx = replaceDockerfileTarWrapper(ctx, buildCtx, relDockerfile, translator, &resolvedTags)
 	}
 
+	// write ctx tar to disk, to make it readable for Image Reference extraction
+	originCtx := buildCtx
+	defer originCtx.Close()
+	readbackTar, err := ioutil.TempFile("", "docker-ctx")
+	defer os.Remove(readbackTar.Name())
```

Comment:
or, in other words - if `ioutil.TempFile` fails, it's nil. [see the example](https://golang.org/pkg/io/ioutil/#TempFile)